subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuildControllers.java

```diff
@@ -68,6 +68,14 @@ public void startTaskExecution() {
     }
 
     @Override
+    public void stopTaskExecution() {
+        for (IncludedBuildController buildController : buildControllers.values()) {
+            buildController.stopTaskExecution();
+        }
+        buildControllers.clear();
+    }
+
+    @Override
     public void stop() {
         CompositeStoppable.stoppable(buildControllers.values()).stop();
```

Comment:
Ah, perhaps this is to `stop` any `IncludedBuildController` that never had `stopTaskExecution` called on it. Will this ever happen?